context/param/methodparams.go

```diff
@@ -0,0 +1,69 @@
+package param
+
+import (
+	"fmt"
+	"strings"
+)
+
+//MethodParam keeps param information to be auto passed to controller methods
+type MethodParam struct {
+	name     string
+	location paramLocation
```

Comment:
http://swagger.io/specification/#parameter-object-44

I would like use the type here. The same as swagger defined. what do you think?